addons/website/static/src/js/content/menu.js

```diff
@@ -142,6 +142,11 @@ publicWidget.registry.autohideMenu = publicWidget.Widget.extend({
                 dom.initAutoMoreMenu(self.$el, {unfoldable: '.divider, .divider ~ li'});
             }
             self.$el.removeClass('o_menu_loading');
+            var header = $('.o_header_freeze');
+            if (header) {
+                $('main').css('padding-top', header.height());
+                header.addClass('o_header_affix affixed');
+            }
```

Comment:
This should be in another dedicated widget like you did for the other header effects. At the start of the new widget, check if the o_menu_loading class is already gone. If it is the case, add you classes and margin-top. If not the case, wait for a special event before doing it. Trigger that special event here.

This is important because your new widget should implement its `destroy` function which removes the padding-top you set here and restores the potential one there already is. The `destroy` is called when entering and leaving edit mode for example.